src/js/player.js

```diff
@@ -450,13 +602,66 @@ class Player extends Component {
     } else if (!this.paused()) {
       this.pause();
     }
+
+    this.trigger('ended');
   }
 
   /**
    * Fired when the duration of the media resource is first known or changed
    * @event durationchange
    */
-  handleDurationChange() {
+  handleTechDurationChange() {
```

Comment:
speaking of which, do you think this change would allow us to update the control bar with the duration before we hit play? https://github.com/videojs/video.js/blob/v4.12.5/src/js/control-bar/time-display.js#L43-L55